src/analysis.ts

```diff
@@ -100,8 +100,9 @@ export function utonalFundamental(frequencies: number[], maxDivisor = 23) {
   return divisor * frequencies[0]
 }
 
-// Interval matrix a.k.a the modes of a scale
-export function intervalMatrix(scale: Scale) {
+// Interval matrix a.k.a the modes of a scale.
+export function intervalMatrix(scale: Scale): [[Interval]] {
```

Comment:
This is not the correct type syntax.
![image](https://github.com/xenharmonic-devs/scale-workshop/assets/1253499/1c51b26b-713c-48a5-b28f-8cf1c8784ecc)
Please use an editor with a working TypeScript plugin.

Reply:
I think it's because I didn't import Interval. I should remove that declaration.

Reply:
Then it's a valid type, but still the wrong one. The return type seems to be `Interval[][]`.

Reply:
That was syntax used in Haskell.

Reply:
I just use Linux Mint's default text editor. The npm didn't catch any type errors, and I should have asked why.

Reply:
`npm run type-check` does indeed pass. That's on Vue for writing a poor variant of TS.